hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/BulkDeleteCommand.java

```diff
@@ -63,6 +88,13 @@ protected LinkedList<PathData> expandArguments(LinkedList<String> args) throws I
         return pathData;
     }
 
+    void deleteInBatches(BulkDelete bulkDelete, List<Path> paths) throws IOException {
+        Batch<Path> batches = new Batch<>(paths, pageSize);
+        while(batches.hasNext()) {
+            bulkDelete.bulkDelete(batches.next());
```

Comment:
Use try catch and print the failed ones if it failed half way.

Reply:
What would be exception thrown in that case?

Reply:
IOE